lib/packmatic/encoder/encoding_state.ex

```diff
@@ -13,6 +13,7 @@ defmodule Packmatic.Encoder.EncodingState do
           remaining: [Entry.t()],
           zstream: nil | :zlib.zstream(),
           bytes_emitted: non_neg_integer(),
+          options: Keyword.t(),
```

Comment:
If compression_level is the only key and there are fixed atoms, it should be typed, aka

```
@type option :: {:compression_level, :zlib.zlevel()}` 
```

Also did you see zstrategy? Looks new

I would in fact make this a map and inherit from default aka

```
@default_options %{compression_level: :default}
```

Then

```
{on_event, options} = Keyword.pop(options, :on_event)
…
{zlib_options, options} = Keyword.pop(options, :zlib_options)
```

then use `zlib_options` as the key and not to use `:options` in this struct anymore so, it is overall clearer b/c otherwise we have options in options and it becomes unclear

The other solution is to simply add a member called `:compression_level` which is also equally clear